budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/parse_exemption_data.py

```diff
@@ -0,0 +1,108 @@
+        {"name": "start_date", "type": "date", "required": True},
+        {"name": "end_date", "type": "date", "required": True},
+        {"name": "decision", "type": "string", "required": True},
+        {"name": "page_title", "type": "string", "required": True},
+        # documents should be in a related table (there could be multiple documents per exemption)
+        {"name": DOCUMENTS_JSON_FIELD, "type": "string", "required": True}
+    ]
+}
+BASE_URL = "http://www.mr.gov.il"
+INPUT_FIELDS_TEXT_MAP = {
+    "publication_id": "SERIAL_NUMBER",
+    "description": "PublicationName",
+    "supplier_id": "SupplierNum",
+    "supplier": "SupplierName",
+    "contact": "ContactPersonName",
+    "publisher": "PUBLISHER",
+    "contact_email": "ContactPersonEmail",
+    "claim_date": "ClaimDate",
+    "last_update_date": "UpdateDate",
+    "reason": "PtorReason",
+    "source_currency": "Currency",
+    "regulation": "Regulation",
+    "volume": "TotalAmount",
+    "subjects": "PublicationSUBJECT",
+    "start_date": "StartDate",
+    "end_date": "EndDate",
+    "decision": "Decision",
+    "page_title": "PublicationType",
+}
+
+
+class ParseExemptionDataProcessor(ResourceFilterProcessor):
+
+    def __init__(self, **kwargs):
+        self._base_url = kwargs.pop("base_url", BASE_URL)
+        super(ParseExemptionDataProcessor, self).__init__(default_input_resource=INPUT_RESOURCE,
+                                                          default_output_resource=OUTPUT_RESOURCE,
+                                                          default_replace_resource=True,
+                                                          table_schema=TABLE_SCHEMA,
+                                                          **kwargs)
+
+    def filter_resource_data(self, data, parameters):
+        for exemption in data:
+            page = pq(exemption["data"])
+            documents = []
+            for update_time_elt, link_elt, img_elt in zip(page("#ctl00_PlaceHolderMain_pnl_Files .DLFUpdateDate"),
+                                                          page("#ctl00_PlaceHolderMain_pnl_Files .MrDLFFileData a"),
+                                                          page("#ctl00_PlaceHolderMain_pnl_Files .MrDLFFileData img")):
+                documents.append({"description": img_elt.attrib.get("alt", ""),
+                                  "link": "{}{}".format(BASE_URL, link_elt.attrib.get("href", "")),
+                                  "update_time": update_time_elt.text,})
+            source_data = {
+                k: page("#ctl00_PlaceHolderMain_lbl_{}".format(v)).text() for k, v in INPUT_FIELDS_TEXT_MAP.items()}
+            yield {
+                "publisher_id": int(exemption["pid"]),
+                "publication_id": int(source_data["publication_id"]),
+                "page_url": exemption["url"],
+                "description": source_data["description"],
+                "supplier_id": int(source_data["supplier_id"]),
```

Comment:
I couldn't find any examples of non-numeric supplier id.. both in the legacy DB and in the data I get from scraping

`select * from exemption where supplier_id is not null and supplier_id !~ '^([0-9]{4,9})$' limit 2`
http://data.obudget.org/queries/1123/source#table